src/particles/ImpactXParticleContainer.cpp

```diff
@@ -100,8 +102,8 @@ namespace impactx
         particle_tile.resize(new_np);
         amrex::copyParticles(
                 particle_tile, pinned_tile, 0, old_np, pinned_tile.numParticles());
-//        Redistribute(); // TODO
 
+        Redistribute();
```

Comment:
One thing why we commented this out here for now:

I think we have not set up our box yet to a reasonable extent. The box extent is determined from the positions of the min/max particle positions in 3D.

Maybe we can discuss with @qianglbl @cemitch99 @RemiLehe how to solve this chicken-egg problem in a clever way.

Reply:
We discussed this in our developer meeting last week.

It would be good to solve the "box extent" chicken-egg problem by:
- optionally asking the user for an initial bounding box (if we initialize in parallel from arbitrary distributions)

Later on, in case we load beams from a data file, we can ask the data file about its min/max values.

Reply:
We currently do that after they are created, as part of the PIC loop:
https://github.com/ECP-WarpX/impactx/blob/570abd95e167f03479826a0bd21ecbaf509fb555/src/ImpactX.cpp#L172-L173
https://github.com/ECP-WarpX/impactx/blob/570abd95e167f03479826a0bd21ecbaf509fb555/src/ImpactX.cpp#L135-L151

Do you suggest we could also do it *while* we initialize them?

Reply:
We currently do that after they are created, as part of the PIC loop:
https://github.com/ECP-WarpX/impactx/blob/570abd95e167f03479826a0bd21ecbaf509fb555/src/ImpactX.cpp#L172-L173
https://github.com/ECP-WarpX/impactx/blob/570abd95e167f03479826a0bd21ecbaf509fb555/src/ImpactX.cpp#L135-L151

Do you suggest we could also do it *while* we initialize them?

Reply:
Yes, that's what I mean - I'll update this PR to do so.